frontend/src/views/CalculateNutrients/CalculateNutrients.tsx

```diff
@@ -104,7 +126,17 @@ export default function CalculateNutrients() {
     const handleEditRow = (e: { id: GridRowId; api: GridApiCommunity }) => {
       setOpenDialog(['fertilizer', e.api.getRowIndexRelativeToVisibleRows(e.id)]);
     };
-    const handleDeleteRow = genHandleDeleteRow(activeField, 'fertilizers', setFieldList);
```

Comment:
You could alter `genHandleDeleteRow` to use this new logic so we don't have essentially the same function written out four times.

Reply:
I recall trying that for `genHandleDeleteRow` but certain array manipulations prevented it from being possible.